### PR TITLE
Separate points by categories

### LINKED ISSUE #115: Category-specific elo
As pointed out by the community, different scored categories require vastly different skillsets (especially LP). It would make sense to rank categories on their own.

### PR DESCRIPTION
**Pull request by p2r3**
The Elo system as a whole is left unchanged, but individual categories are now treated as having their own leaderboards. This pull request implements both the frontend and backend code for this change. <sub>(Yes, all in one commit.)</sub>

Closes #115

### CODE REVIEW

**Review by PancakeTAS — APPROVED** [review]
untested but code looks good

File: util/points.js [hunk 1]
```diff
@@ -123,20 +119,22 @@ function calculateDisplayPoints (statistics) {
...
+  if (!(category in profile.statistics)) profile.statistics[category] = [];
+  return calculateTotalPoints(profile.statistics[category]);
 
 }
 
 /**
  * Calculate the elo acquired or lost for each player in this week.
  * Note: Despite only returning the elo delta, the calculation itself is
- * based on the elo already found in the profile data (see pointsFromSteamID).
+ * based on the elo already found in the profile data (see pointsFromSteamID)., catname
```
Comment (soni801) [inline util/points.js:137]: What is going on here?

**Review by soni801 — APPROVED** [review]
I don't really know how all the points stuff works, but nothing stood out as looking broken. I also haven't tested but it's proooooobably gonna be fine :Clueless:

File: util/points.js (hunk 1 above)
Comment (p2r3) [reply util/points.js:137]: whoops.

**Pull request merged**